outputs.tf

```diff
@@ -20,12 +20,13 @@ output "list" {
   value = [
     for vsi_key, virtual_server in ibm_is_instance.vsi :
     {
-      name         = virtual_server.name
-      id           = virtual_server.id
-      zone         = virtual_server.zone
-      ipv4_address = virtual_server.primary_network_interface[0].primary_ipv4_address
-      floating_ip  = var.enable_floating_ip ? ibm_is_floating_ip.vsi_fip[vsi_key].address : null
-      vpc_id       = var.vpc_id
+      name                   = virtual_server.name
+      id                     = virtual_server.id
+      zone                   = virtual_server.zone
+      ipv4_address           = virtual_server.primary_network_interface[0].primary_ipv4_address
+      secondary_ipv4_address = length(virtual_server.network_interfaces) == 0 ? "" : virtual_server.network_interfaces[0].primary_ipv4_address
```

Comment:
Is there a reason you default to empty string instead of null when the length is 0?